transcoders.go

```diff
@@ -121,61 +131,36 @@ func onionBtS(b []byte) (string, error) {
 	return addr + ":" + strconv.Itoa(int(port)), nil
 }
 
-var TranscoderP2P = NewTranscoderFromFunctions(p2pStB, p2pBtS)
+var TranscoderP2P = NewTranscoderFromFunctions(p2pStB, p2pBtS, p2pVal)
 
 func p2pStB(s string) ([]byte, error) {
 	// the address is a varint prefixed multihash string representation
 	m, err := mh.FromB58String(s)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse p2p addr: %s %s", s, err)
 	}
-	size := CodeToVarint(len(m))
-	b := append(size, m...)
-	return b, nil
+	return m, nil
 }
 
-func p2pBtS(b []byte) (string, error) {
-	// the address is a varint-prefixed multihash string representation
-	size, n, err := ReadVarintCode(b)
-	if err != nil {
-		return "", err
-	}
+func p2pVal(b []byte) error {
+	_, err := mh.Cast(b)
+	return err
+}
 
-	b = b[n:]
-	if len(b) != size {
-		return "", errors.New("inconsistent lengths")
-	}
+func p2pBtS(b []byte) (string, error) {
 	m, err := mh.Cast(b)
 	if err != nil {
 		return "", err
 	}
 	return m.B58String(), nil
 }
 
-var TranscoderUnix = NewTranscoderFromFunctions(unixStB, unixBtS)
+var TranscoderUnix = NewTranscoderFromFunctions(unixStB, unixBtS, nil)
 
 func unixStB(s string) ([]byte, error) {
-	// the address is the whole remaining string, prefixed by a varint len
-	size := CodeToVarint(len(s))
-	b := append(size, []byte(s)...)
-	return b, nil
+	return []byte(s), nil
```

Comment:
are we dropping the varints for some reason? I assume youre not changing the binary format, so where is this being moved to?

Reply:
We're now doing this in stringToBytes and bytesToString. Before, we were doing this for *every* protocol with a variable length size (and we were doing the exact same thing. On the other hand, we had *other* code that assumed that we were using varint length prefixes so I figured we might as well just centralize this logic and simplify things.

This change and the path one are both to pave the way for nice multiaddr constructors/decoders (so we can stop parsing strings).